yabt/buildcontext.py

```diff
@@ -453,37 +503,14 @@ def build(target: Target):
                     logger.info('Build of target {} completed in {} sec',
                                 target.name, target.summary['build_time'])
                     target_built = True
-                built_targets.add(target.name)
-                target.done()
 
-                # TODO: attempt flaky tests N times AFTER all the rest passed
                 # TODO: collect stats and print report at the end
-                # TODO: support both "fail fast" (exit on first failure) and
-                #       run all tests (don't exit on first failure) modes
                 target_tested = False
                 if run_tests and 'testable' in target.tags:
                     if self.conf.no_test_cache or not test_cached:
                         logger.info('Testing target {}', target.name)
-                        attempts = (
-                            self.conf.test_attempts if
-                            target.props.attempts == 1 else
-                            target.props.attempts)
-                        if attempts < 1:
-                            raise ValueError(
-                                'Attempts value must be 1 or more: got {}'
-                                .format(attempts))
-                        test_start = 0
-                        target.info['fail_count'] = 0
-                        while attempts > target.info['fail_count']:
-                            test_start = time()
-                            try:
-                                self.test_target(target)
-                            except Exception:
-                                target.info['fail_count'] += 1
-                                if attempts <= target.info['fail_count']:
-                                    raise
-                            else:
-                                break
+                        test_start = time()
+                        self.test_target(target)
                         target.info['test_time'] = time() - test_start
                         logger.info(
```

Comment:
is it still relevant to log `fail_count` here given the changes?

Reply:
Yeah, that's the only place where we document the flakiness of a test.
I can move this print to the bottom. But this is not a failed target, it's a target the passed on its n-th trial.

Reply:
your call